module/VuFind/src/VuFind/ILS/Driver/XCNCIP2.php

```diff
@@ -1139,33 +1303,39 @@ public function getMyStorageRetrievalRequests($patron)
                 'ns1:Ext/ns1:BibliographicDescription/' .
                 'ns1:BibliographicRecordId/ns1:AgencyId'
             );
-            //$created = $current->xpath('ns1:DatePlaced');
             $title = $current->xpath('ns1:Title');
             $pos = $current->xpath('ns1:HoldQueuePosition');
             $pickupLocation = $current->xpath('ns1:PickupLocation');
             $requestId = $current->xpath('ns1:RequestId/ns1:RequestIdentifierValue');
             $requestType = $current->xpath('ns1:RequestType');
             $requestType = (string)$requestType[0];
-            $tmpStatus = $current->xpath('ns1:RequestStatusType');
-            list($status, $created) = explode(" ", (string)$tmpStatus[0], 2);
-            if ($status === "Accepted") {
-                $cancelled = false;
+            $created = $current->xpath('ns1:DatePlaced');
+            $created = $this->displayDate(
+                !empty($created) ? (string)$created[0] : null
+            );
+            $requestStatusType = $current->xpath('ns1:RequestStatusType');
+            $status = !empty($requestStatusType) ? (string)$requestStatusType[0]
+                : null;
+
+            $processed = false;
+            if ($status === 'Available For Pickup') {
```

Comment:
Should this be case-insensitive and/or defined in a class property?

Reply:
Done. I even ended up with refactoring of getMyHolds and getMyStorageRetrievalRequests methods... ;)